website/src/app/blog/devlog/2025-09/readme.mdx

```diff
@@ -0,0 +1,163 @@
+Proxy IP assignments for DNS resources now persist across Client sessions.[^10104]
+Previously, restarting the Client could reassign `100.96.0.1` to a different resource, breaking applications that cached the IP address.
+Now these mappings stay consistent for the lifetime of the tunnel process, fixing issues with long-lived connections like SSH sessions.
+
+## Better DNS Resolution
+
+The Gateway switched from using the system's `libc` resolver to the `hickory-resolver` library for A and AAAA record resolution.[^10373]
+This change enables proper TTL-based caching instead of the previous hardcoded 30-second cache.
+DNS responses are now cached for exactly as long as the authoritative nameserver specifies, reducing unnecessary queries while ensuring timely updates.
+
+## Performance Under Load
+
+We optimized the event-loop to prevent starvation of lower-priority inputs.[^10347]
+The tunnel now batch-processes input from all sources rather than prioritizing high-traffic sources to completion.
+This prevents timeout checks and DNS resolution from being delayed when UDP sockets or the TUN device are extremely busy.
+
+Linux systems got a significant performance boost through better socket buffer management.[^10349]
+Clients and Gateways now attempt to set receive and send buffer maximums to 4MB at startup.
+The default 200KB buffers were causing packet drops during high throughput scenarios, directly correlating with buffer errors visible in `nstat`.
+
+We also limited the number of optimistic ICE candidates to prevent CPU spikes.[^10367]
+When clients advertise many IPv6 addresses, the previous unlimited candidate generation could cause performance issues.
+Optimistic candidates are now disabled entirely for IPv6 and limited to 2 for IPv4, focusing on the scenarios where they're most beneficial.
+
+## Relay Infrastructure Advances
+
+Our relay infrastructure received important eBPF improvements.
+The relay can now handle relay-to-relay candidate pairs, which occur when both Client and Gateway allocate from the same relay.[^10286]
+Previously these packets would need to traverse userspace or fail entirely.
+
+We restructured the eBPF code into modular components that handle all cross-stack translation cases, including IPv4↔IPv6 transformations.
+A comprehensive integration test with double symmetric NAT validates this behavior in continuous integration.
+
+We also fixed several edge cases in the eBPF layer, including properly handling DNS replies[^10330] and re-populating the channel map when TURN channels are refreshed.[^10291]
+
+## Platform-Specific Improvements
+
+The Android client now launches authentication in a CustomTab instead of the default browser.[^10371]
+This fixes a Firefox bug where only the first browser tab could intercept the custom URI scheme, making subsequent sign-ins fail until users manually closed the old tab.
+CustomTabs ensure only one sandboxed authentication instance exists at a time.
```

Comment:
```suggestion
CustomTabs ensure only one sandboxed authentication instance exists at a time.
It also results in a nice UX improvement where the "You have been signed-in" tabs are no longer lingering around.
```

I think that is quite nice. It would also be awesome if we could achieve the same thing on desktop somehow.